lib/engine/step/g_18_co/tracker.rb

```diff
@@ -25,6 +25,15 @@ def collect_mines(corporation, hex)
           @log << "#{corporation.name} collects a mine token from #{hex.name}"
         end
 
+        def lay_tile(action, extra_cost: 0, entity: nil, spender: nil)
+          entity ||= action.entity
+          @previous_connections = prior_connected_paths(entity, action.hex.tile)
```

Comment:
looks like this calls the graph outside of game load, this will kill performance

Reply:
> looks like this calls the graph outside of game load, this will kill performance

I don't see how I can avoid checking the graph ( the previous implementation also used the graph ) but previously it only checked on a very limited set of tiles. In order to get back to that state, I would have to to duplicate the code in both the base check_track_restrictions! as well as 18COs check_track_restrictions!.

Do you have a different suggestion for how I can find out the reachable segments of track on a hex for an entity before a tile is laid?

Reply:
can you only check this when !@game.loading

Reply:
Why would this kill performance anyway? Its not regenerating the graph. Base `check_track_restrictions!` calls the same `@game.graph.connected_paths(entity)` repeatedly.

Reply:
only when !game.loading